recon/src/recon/pending_cache.rs

```diff
@@ -0,0 +1,153 @@
+use std::collections::HashMap;
+
+use crate::{Key, PendingItem, ReconItem};
+
+#[derive(Debug, Default)]
+/// This struct manages tracking items that we attempted to deliver but could not be processed
+/// because we need to discover additional information.
+pub struct PendingCache<K: Key> {
+    by_needed_key: HashMap<Vec<u8>, ReconItem<K>>,
```

Comment:
This needs to point to a vec of items as we could have many events depending on one init event. However, we may nuke this concept and just keep it really simple initially, simply returning the items and putting them in the next batch.